File: java/src/test/java/io/github/garudadidada/HumanizeTest.java
package io.github.garudadidada;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class HumanizeTest {

    @Test
    void formatBytes() {
        assertEquals("0 B", Humanize.formatBytes(0));
        assertEquals("500 B", Humanize.formatBytes(500));
        assertEquals("1 KB", Humanize.formatBytes(1024));
        assertEquals("1.5 KB", Humanize.formatBytes(1536));
        assertEquals("1 MB", Humanize.formatBytes(1048576));
        assertEquals("1 GB", Humanize.formatBytes(1073741824L));
    }

    @Test
    void formatDuration() {
        assertEquals("0s", Humanize.formatDuration(0));
        assertEquals("45s", Humanize.formatDuration(45));
        assertEquals("1m 30s", Humanize.formatDuration(90));
        assertEquals("1h 1m 1s", Humanize.formatDuration(3661));
        assertEquals("1d 1h 1m 1s", Humanize.formatDuration(90061));
    }

    @Test
    void ordinal() {
        assertEquals("1st", Humanize.ordinal(1));
        assertEquals("12th", Humanize.ordinal(12));
        assertEquals("21st", Humanize.ordinal(21));
        assertEquals("22nd", Humanize.ordinal(22));
        assertEquals("101st", Humanize.ordinal(101));
        assertEquals("111th", Humanize.ordinal(111));
    }

    @Test
    void pluralize() {
        assertEquals("1 file", Humanize.pluralize(1, "file"));
        assertEquals("3 files", Humanize.pluralize(3, "file"));
        assertEquals("2 people", Humanize.pluralize(2, "person", "people"));
    }
}

File: php/tests/HumanizeTest.php
<?php

declare(strict_types=1);

require __DIR__ . '/../src/Humanize.php';

use Garudadidada\Humanize;

$failures = 0;
function check($got, string $want, string $label): void
{
    global $failures;
    if ($got !== $want) {
        fwrite(STDERR, "FAIL $label: got '" . var_export($got, true) . "' want '$want'\n");
        $failures++;
        return;
    }
    echo "ok  $label\n";
}

check(Humanize::formatBytes(0), '0 B', 'bytes(0)');
check(Humanize::formatBytes(500), '500 B', 'bytes(500)');
check(Humanize::formatBytes(1024), '1 KB', 'bytes(1024)');
check(Humanize::formatBytes(1536), '1.5 KB', 'bytes(1536)');
check(Humanize::formatBytes(1048576), '1 MB', 'bytes(1048576)');
check(Humanize::formatBytes(1073741824), '1 GB', 'bytes(1GB)');

check(Humanize::formatDuration(0), '0s', 'dur(0)');
check(Humanize::formatDuration(45), '45s', 'dur(45)');
check(Humanize::formatDuration(90), '1m 30s', 'dur(90)');
check(Humanize::formatDuration(3661), '1h 1m 1s', 'dur(3661)');
check(Humanize::formatDuration(90061), '1d 1h 1m 1s', 'dur(90061)');

check(Humanize::ordinal(1), '1st', 'ord(1)');
check(Humanize::ordinal(2), '2nd', 'ord(2)');
check(Humanize::ordinal(3), '3rd', 'ord(3)');
check(Humanize::ordinal(11), '11th', 'ord(11)');
check(Humanize::ordinal(12), '12th', 'ord(12)');
check(Humanize::ordinal(13), '13th', 'ord(13)');
check(Humanize::ordinal(21), '21st', 'ord(21)');
check(Humanize::ordinal(22), '22nd', 'ord(22)');
check(Humanize::ordinal(101), '101st', 'ord(101)');
check(Humanize::ordinal(111), '111th', 'ord(111)');

check(Humanize::pluralize(1, 'file'), '1 file', 'plur(1)');
check(Humanize::pluralize(3, 'file'), '3 files', 'plur(3)');
check(Humanize::pluralize(2, 'person', 'people'), '2 people', 'plur(people)');

if ($failures > 0) {
    fwrite(STDERR, "\n$failures FAILURES\n");
    exit(1);
}
echo "\nALL PHP TESTS PASSED\n";
